youtube_dl/extractor/beatport.py

```diff
@@ -0,0 +1,87 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from youtube_dl.utils import int_or_none
+
+from .common import ExtractorError, InfoExtractor
+
+
+class BeatportIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?beatport\.com/(?P<uploader_id>\w+)/tracks/(?P<id>.+)/(?P<display_id>.+)'
+    _TESTS = [{
+        'url': 'https://www.beatport.com/musicalfreedom/tracks/l2axnqcd6azu/secrets-feat-vassy-original-mix',
+        'md5': '11e330f6236be07f61fcc4669e972cd8',
+        'info_dict': {
+            'id': 'l2axnqcd6azu',
+            'display_id': 'secrets-feat-vassy-original-mix',
+
+            'uploader': 'Musical Freedom',
+            'uploader_id': 'musicalfreedom',
+            'uploader_url': 'http://www.musicalfreedom.com',
+
+            'ext': 'mp4',
+            'title': 'Secrets feat. Vassy',
+            'duration': 267,
+            'artist': 'Tiësto, VASSY, KSHMR',
+            'genre': 'Electro House',
+            'release_year': '2015',
+            'thumbnail': 'https://api-media.beatport.com/sounds/l2axnqcd6azu/1431513641.jpg',
+
+            'view_count': int,
+            'like_count': int
+        }
+    }, {
+        'url': 'https://www.beatport.com/flamingorecordingsbps/tracks/qpwhxukynk7r/burnin-up-original-mix',
+        'md5': '67bb3421c43f92b5a625c2065255925f',
+        'info_dict': {
+            'id': 'qpwhxukynk7r',
+            'display_id': 'burnin-up-original-mix',
+
+            'uploader': 'Flamingo Recordings',
+            'uploader_id': 'flamingorecordingsbps',
+            'uploader_url': 'http://www.flamingorecordings.com',
+
+            'ext': 'mp4',
+            'title': "Burnin' Up",
+            'duration': 272,
+            'artist': 'Flatdisk',
+            'genre': 'Progressive House',
+            'release_year': '2015',
+            'thumbnail': 'https://api-media.beatport.com/sounds/qpwhxukynk7r/1430858737.jpg',
+
+            'view_count': int,
+            'like_count': int
+        }
+    }
+    ]
+
+    def _real_extract(self, url):
+        url_data = self._VALID_URL_RE.match(url)
+        api_base = 'https://apix.beatport.com'
+        api_data = self._download_json('%s/sounds/%s?format=json' % (api_base, url_data.group('id')),
```

Comment:
You are calculating `url_data.group('id')` over and over again. Extract it into a variable.